modules/tinymce/src/core/main/ts/fmt/ExpandRange.ts

```diff
@@ -190,10 +183,12 @@ const findParentContainer = (
     if (!format[0].block_expand && dom.isBlock(parent)) {
       return parent;
     }
-
     // Walk left/right
-    for (sibling = parent[siblingName]; sibling; sibling = sibling[siblingName]) {
-      if (!isBookmarkNode(sibling) && !isWhiteSpaceNode(sibling) && !isBogusBr(sibling)) {
+    for (sibling = parent[siblingName] as Element; sibling; sibling = sibling[siblingName] as Element) {
+      const siblingIsNotSpace = !(sibling as any).data || (sibling as any).data !== ' ';
```

Comment:
Shouldn't sibling be of type Node? Then you don't need to cast as element and you could do `const siblingIsSpace = NodeType.isText(sibling) && sibling.data === ' ';` here.

Reply:
No, because isBogusBr requires an element, and Node does not have the data-property.
Honestly whatever is picked will require casting. It's somewhat annoying.

Reply:
Honesty, this sounds a bit like a solution in search of a problem. The thing here is that the Node-types just return Node, even if what it really is is far higher up the ladder. Doing these overly complicated optional-things seems like it'd honestly just make things a lot harder to read, with very little benefit.
I'll be honest and say that I personally think that casting is perfectly fine in this one, but for a professional setting I suppose I can see why we might want to use type-guards, even if it makes the code less clear. I'll do something with that.